homework_5

```diff
@@ -0,0 +1,262 @@
+kill %1
+# [1]   Terminated              sleep 100000
+
+
+
+# 5. To other send a SIGCONT in two different ways.
+# 1)
+kill -n 18 %2
+# 2)
+kill -SIGCONT 1369
+# можно было бы воспользоваться командой pkill - она бы запустила все процессы sleep, что по сути в моей ситуации являлось запуском только одной из двух оставшихся, но это как-то неправильно. Получается, pkill в некотором плане менее гибкая, так как будет работать со всеми процессами с одинаковым названием
+
+
+
+# 6. Kill one by PID and the second one by job ID
+kill 1368
+# [2]-  Terminated              sleep 500000
+# [3]+  Running                 sleep 900000 &
+kill %3
+# [3]+  Terminated              sleep 900000
+
+
+
+# systemd
+# 1. Write two daemons: one should be a simple daemon and do sleep 10 after a start and
+# then do echo 1 > /tmp/homework, the second one should be oneshot and do echo 2 >
+# /tmp/homework without any sleep
+
+# Демона нужно создавать в директории /etc/systemd/systemd
+# Про создание юнита демона можно почитать man SYSTEMD.UNIT
+# В лекции было упомянуто отдельно узнать про type, про него можно почитать man SYSTEMD.SERVICE
+# Напишу пару скриптов согласно заданию, расположу их в директории своего пользователя /home/anme/
+# script1.sh:
+  # #!/bin/bash
+  # sleep 10
+  # echo 1>/tmp/homework
+# script2.sh:
+  # #!/bin/bash
+  # 2>/tmp/homework
+# в директории /etc/systemd/system создам файл daemonone.service
+# daemonone.service:
+  # [Unit]
+  # Description=My Daemon One with sleep command
+  # [Service]
+  # Type=simple
+  # ExecStart=/home/anme/script1.sh
+  # [Install]
+  # WantedBy=multi-user.target
+# уже можно проверить статус моего демона
+systemctl status daemonone.service
+# ● daemonone.service - My Daemon One with sleep command
+#    Loaded: loaded (/etc/systemd/system/daemononed.service; disabled; vendor preset: disabled)
+#    Active: inactive (dead)
+
+
+
+# 2. Make the second depended on the first one (should start only after the first)
+# в директории /etc/systemd/system создам файл daemontwo.service
+# daemontwo.service:
+  # [Unit]
+  # Description=My Daemon Two without sleep command
+  # After=daemonone.service
+  # [Service]
+  # ExecStart=/home/anme/script2.sh
+  # [Install]
+  # WantedBy=multi-user.target
+
+
+
+# 3. Write a timer for the second one and configure it to run on 01.01.2019 at 00:00
+# Судя по всему имеется ввиду еще один юнит .timer. Напишем такой
+# daemontwo.timer:
+  # [Unit]
+  # Description=Timer for Daemon Two
+  # [Timer]
+  # Unit=daemontwo.service
+  # OnCalendar=2019-01-01 00:00
+  # [Install]
+  # WantedBy=timers.target
+
+
+
+# 4. Start all daemons and timer, check their statuses, timer list and /tmp/homework
+sudo systemctl enable daemonone.service
+sudo systemctl enable daemontwo.service
+sudo systemctl enable daemontwo.timer
+sudo chmod 777 /home/anme/* # не сразу заметил, что у демона не будет разрешения на исполнение. Для надежности сразу же выдал полные права на оба скрипта в дирекории
```

Comment:
chmod 777 - очень плохая практика; следует его использовать только как один из шагов при дебаге. 

А из под какого пользователя запускались демоны? Были ли фактические проблемы с доступом к скриптам?

Reply:
Демонов запускал от рута.
Вместо chmod 777 достаточно сделать 
```
[anme@localhost ~]$ chmod u+x ~/script1.sh
[anme@localhost ~]$ chmod u+x ~/script2.sh
```
т.е. выдать права на исполнение на эти скрипты
Попробовал теперь запускать демона от своей учетки anme

`[anme@localhost ~]$ systemctl start daemonone.service`
==== AUTHENTICATING FOR org.freedesktop.systemd1.manage-units ===
Authentication is required to manage system services or units.
Authenticating as: anme
Password:
==== AUTHENTICATION COMPLETE ===
`[anme@localhost ~]$ systemctl status daemonone.service`
● daemonone.service - My Daemon One with sleep command
   **Loaded: loaded (/etc/systemd/system/daemonone.service; enabled; vendor preset: disabled)
   Active: active (running) since Thu 2021-12-23 06:10:06 CST; 1s ago**
 Main PID: 2061 (script1.sh)
   CGroup: /system.slice/daemonone.service
           ├─2061 /bin/bash /home/anme/script1.sh
           └─2062 sleep 10

Dec 23 06:10:06 localhost.localdomain systemd[1]: Started My Daemon One with sleep command.
[anme@localhost ~]$ systemctl status daemonone.service
● daemonone.service - My Daemon One with sleep command
   **Loaded: loaded (/etc/systemd/system/daemonone.service; enabled; vendor preset: disabled)
   Active: inactive (dead) since Thu 2021-12-23 06:10:16 CST; 45s ago**
  Process: 2061 ExecStart=/home/anme/script1.sh (code=exited, status=0/SUCCESS)
 Main PID: 2061 (code=exited, status=0/SUCCESS)

Dec 23 06:10:06 localhost.localdomain systemd[1]: Started My Daemon One with sleep command.
[anme@localhost ~]$